qutip/bloch.py

```diff
@@ -415,29 +415,33 @@ def add_vectors(self, vectors, colors=None, alpha=1.0):
             Transparency value for the vectors. Values between 0 and 1.
 
         """
-        if isinstance(vectors[0], (list, tuple, np.ndarray)):
-            if colors is not None:
-                for k, vec in enumerate(vectors):
-                    self.vectors.append(vec)
-                    self.vector_color.append(colors[k])
-                    self.vector_alpha.append(alpha)
-            else:
-                for vec in vectors:
-                    k = np.mod(len(self.vectors),
-                               len(self.vector_default_color))
-                    color = self.vector_default_color[k]
-                    self.vector_color.append(color)
-                    self.vectors.append(vec)
-                    self.vector_alpha.append(alpha)
+        vectors = np.asarray(vectors)
+
+        if vectors.ndim == 1:
+            vectors = vectors[np.newaxis, :]
+
+        if vectors.ndim != 2 or vectors.shape[1] != 3:
+            raise ValueError(
+                "The included vectors are not valid. Vectors must "
+                "be equivalent to a 2D array where the first "
+                "index represents the iteration over the vectors and the "
+                "second index represents the position in 3D of vector head.")
+
+        n_vectors = vectors.shape[0]
+        if colors is None:
+            colors = np.array([None]*n_vectors)
```

Comment:
```suggestion
            colors = np.array([None] * n_vectors)
```

Reply:
Maybe we should allow colors to be a scalar and then expand that to all vectors?

Reply:
Yes definitely. If you do not mind though, I'd rather do that in another PR once I have a better idea of how I would like to implement the kw arguments for plotting. There are in fact some challenges with points using two different methods for plotting and some arguments being attributes in the bloch sphere. I will try to address all of that in a single PR. For the moment, this fixes a few things and it already adds some small features.

Reply:
Sound good.